### PR TITLE
Add section on token credentials to SQL Gateway

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

**Comment by andreasohlund** [comment]
FYI SqlT targets the new client only

https://github.com/Particular/NServiceBus.SqlServer/blob/master/src/NServiceBus.Transport.SqlServer/NServiceBus.Transport.SqlServer.csproj#L15

File: nservicebus/gateway/sql/index.md [hunk 1]
```diff
@@ -17,6 +17,13 @@ By default, the component uses a default schema and table name of `[dbo].[Gatewa
 
 snippet: DefaultUsage
 
+### Token-credentials
+
+Microsoft Entra authentication is supported via the [standard connection string options](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication).
```
Comment (PhilBastian) [inline nservicebus/gateway/sql/index.md:22]:
```suggestion
Microsoft Entra Id authentication is supported via the [standard connection string options](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication).
```

File: nservicebus/gateway/sql/index.md [hunk 2]
```diff
@@ -17,6 +17,13 @@ By default, the component uses a default schema and table name of `[dbo].[Gatewa
...
+
+> [!NOTE]
+> Entra authentication is only supported when using [Microsoft.Data.SqlClient](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#overview)
```
Comment (PhilBastian) [inline nservicebus/gateway/sql/index.md:25]:
```suggestion
> Entra Id authentication is only supported when using [Microsoft.Data.SqlClient](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#overview)
```

File: persistence/sql/dialect-mssql.md [hunk 3]
```diff
@@ -26,6 +26,9 @@ include: mssql-dtc-warning
 
 Microsoft Entra authentication is supported via the [standard connection string options](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication).
 
+> [!NOTE]
+> Entra authentication is only supported when using [Microsoft.Data.SqlClient](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#overview)
```
Comment (PhilBastian) [inline persistence/sql/dialect-mssql.md:30]:
```suggestion
> Entra Id authentication is only supported when using [Microsoft.Data.SqlClient](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#overview)
```

**Review by PhilBastian — APPROVED** [review]

**Comment by danielmarbach** [comment]
LGTM

**Comment by adamralph** [comment]
[editorial tweaks](https://github.com/Particular/docs.particular.net/commit/6d785157c6bb997f29a0052389388de46b64709e)

**Pull request merged**